### ISSUE TITLE
Decimal signficance consistency through sessions

### ISSUE THREAD

**Issue by tzcrawford** [andmarti1424/sc-im#291]
If you save a csv with cells containing numbers containing a non-default signficance (e.g. 101 rather than 101.00), the file will save with the cells containing the appropriate text (101), but when you open that csv in sc-im again, all numbers will revert to the default significance (101.00) again, and you have to hover back over that column and use f-j or f-k to get it back to the appropriate significance.

**Comment by andmarti1424** [comment]
you are saving in CSV format?

**Comment by tzcrawford** [comment]
Yes. I am saving as filename.csv and I have also tried running scim and using :e csv filename.csv  . The problem still persists. Perhaps I am ignorant on the correct way to save?
________________________________

**Comment by kofm** [comment]
Also, if you write a number i.e. 0.2542 but the format is 10 2 0 in the CSV only 0.25 is saved
So e.g. if you open a CSV file with numbers at the fourth decimal, the number is formatted and then if you save the file, the third and fourth decimal are lost in the process.
Is there any workaround instead of manually adjusting the numbers formatting at each open?

**Comment by andmarti1424** [comment]
You can open and save csv file that has numbers and interpret them as text..
Check the configuration value 'import_delimited_as_text'..

**Comment by andmarti1424** [comment]
@tzcrawford @kofm Was it useful for you?

**Comment by tzcrawford** [comment]
> @tzcrawford @kofm Was it useful for you?
Somewhat.

If I do `:set import_delimited_as_text` and `load file.csv`, or put that in my `.scimrc`, it seems to still open with default formatting `1234.00` as opposed to what it might read in plain text `1234.0`, etc. If I do `sc-im --import_delimited_as_text`, it will open the file with all cells in text format as expected. But then you can't do any math with those cells, and you can't change them into a particular number format as far as I am aware, i.e. with `:format "string"`.

I guess the problem here is that any time you open a csv file with cells containing numbers, it always defaults to the cells to a format with a number and two digits after a decimal (I think). What would be most convenient is if it could somehow guess the format of the cell from the plain-text in the file. For example, if your file read `2.3e5,123.3`, sc-im would automatically format A0 to read `2.3e+5` and B0 to `123.3` instead of `230000.00` and `123.30` respectively. This would be especially convenient in that you don't have to change the formatting of every cell before you save the edited file (and do this procedure every time you restart sc-im loading the input file).

**Comment by kofm** [comment]
I confirm the above. It would not be a big problem except for the fact that if you are not careful you risk deleting data by removing decimal numbers. I agree that the best would be to follow the cell format of the plain text file. 

Thank you so much for your work

**Comment by andmarti1424** [comment]
Yes. It could be an enhacement but the thing is that there are a lot of numeric formats. and it can be tricky and risky.
And slower than just letting sc-im put the numeric value with the default format.
(Some use this to open 60k rows csv's.. Its pretty fast since last updates though)

About the example above, if we try to interpret the format from the file, where file reads  2.3e5,123.3, sc-im should show the same: 2.3e5 and 123.3, both of course as numeric values.
